freqtrade/exchange/okx.py

```diff
@@ -46,3 +47,48 @@ def _lev_prep(
                     "mgnMode": self.margin_mode.value,
                     "posSide": "long" if side == "buy" else "short",
                 })
+
+    def get_max_pair_stake_amount(
+        self,
+        pair: str,
+        price: float,
+        leverage: float = 1.0
+    ) -> float:
+
+        if self.trading_mode == TradingMode.SPOT:
+            return float('inf')  # Not actually inf, but this probably won't matter for SPOT
+
+        if pair not in self._leverage_tiers:
+            return float('inf')
+
+        pair_tiers = self._leverage_tiers[pair]
+        return pair_tiers[-1]['max'] / leverage
+
+    def load_leverage_tiers(self) -> Dict[str, List[Dict]]:
+        # * This is slow(~45s) on Okex, must make 90-some api calls to load all linear swap markets
+        if self.trading_mode == TradingMode.FUTURES:
+            markets = self.markets
+            symbols = []
+
+            for symbol, market in markets.items():
+                if (self.market_is_future(market)
+                        and market['quote'] == self._config['stake_currency']):
+                    symbols.append(symbol)
+
+            tiers: Dict[str, List[Dict]] = {}
+
+            # Be verbose here, as this delays startup by ~1 minute.
+            logger.info(
+                f"Initializing leverage_tiers for {len(symbols)} markets. "
+                "This will take about a minute.")
+
+            for symbol in sorted(symbols):
+                res = self._api.fetch_leverage_tiers(symbol)
+                tiers[symbol] = []
+                for tier in res[symbol]:
+                    tiers[symbol].append(self.parse_leverage_tier(tier))
```

Comment:
I think we'll need to be careful with this
this test fails because we call `parse_leverage_tier()` both here and in `fill_leverage_tiers()`
Obviously parsing the same thing twice on the same object will cause it to fail as we're significantly changing the structure as part of this method.